### PR TITLE
[Fix-10028][Docker]Support Multi-Architecture Docker Images

### LINKED ISSUE #10028: [Bug] [Docker] Docker Build Not Support Multi Architecture
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

When I exec docker build at M1 Mac, it'll get docker image with ```"Architecture": "arm64"```, can't run at Linux amd64.

### What you expected to happen

When running docker build, it can get multi-architecture images with amd64 && arm64.

### How to reproduce

Run at local
```
./mvnw -B clean deploy \
          -Dmaven.test.skip \
          -Dmaven.javadoc.skip \
          -Dmaven.checkstyle.skip \
          -Dmaven.deploy.skip \
          -Ddocker.tag=tag\
          -Ddocker.hub=url\
          -Pdocker,release
```

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by qingwli**
close #10028

### CODE REVIEW

File: docs/docs/en/development/development-environment-setup.md [hunk 1]
```diff
@@ -23,6 +23,24 @@ git clone git@github.com:apache/dolphinscheduler.git
...
+          -Dmaven.test.skip \
+          -Dmaven.javadoc.skip \
+          -Dmaven.checkstyle.skip \
+          -Dmaven.deploy.skip \
+          -Ddocker.tag=<TAG> \
+          -Ddocker.hub=<HUB_URL> \
+          -Pdocker,release          
+```
+
+> **_Note：_** Docker will build linux/amd64,linux/arm64 multi-architecture images by defult
```
Comment (kezhenxu94) [inline docs/docs/en/development/development-environment-setup.md]:
```suggestion
> **_Note：_** Docker will build and push linux/amd64,linux/arm64 multi-architecture images by default
```

File: docs/docs/en/development/development-environment-setup.md [hunk 2]
```diff
@@ -23,6 +23,24 @@ git clone git@github.com:apache/dolphinscheduler.git
...
+$ cd dolphinscheduler
+$ ./mvnw -B clean deploy \
+          -Dmaven.test.skip \
+          -Dmaven.javadoc.skip \
+          -Dmaven.checkstyle.skip \
+          -Dmaven.deploy.skip \
+          -Ddocker.tag=<TAG> \
+          -Ddocker.hub=<HUB_URL> \
+          -Pdocker,release          
+```
```
Comment (kezhenxu94) [inline docs/docs/en/development/development-environment-setup.md]: * To build Docker images locally, run

```shell
$ cd dolphinscheduler
$ ./mvnw -B clean package \
          -Dmaven.test.skip \
          -Dmaven.javadoc.skip \
          -Dmaven.checkstyle.skip \
          -Ddocker.tag=<TAG> \
          -Pdocker,release          
```

* To build and push Docker images to your registry `<HUB_URL>`, run
```shell
$ cd dolphinscheduler
$ ./mvnw -B clean deploy \
          -Dmaven.test.skip \
          -Dmaven.javadoc.skip \
          -Dmaven.checkstyle.skip \
          -Dmaven.deploy.skip \
          -Ddocker.tag=<TAG> \
          -Ddocker.hub=<HUB_URL> \
          -Pdocker,release          
```

Comment (kezhenxu94) [reply docs/docs/en/development/development-environment-setup.md]: @liqingwang please update the doc to 👆

Comment (qingwli) [reply docs/docs/en/development/development-environment-setup.md]: Done

**Comment by zhongjiajie** [comment]
BTW, we already have some content in https://dolphinscheduler.apache.org/en-us/docs/release/faq.html section “How to Build Custom Docker Image”. I think "development" is the better place to hold it, could you migrate them and them delete in FAQ?

File: docs/docs/en/development/development-environment-setup.md [hunk 3]
```diff
@@ -23,6 +23,58 @@ git clone git@github.com:apache/dolphinscheduler.git
...
+       -Dmaven.test.skip \
+       -Dmaven.javadoc.skip \
+       -Dmaven.checkstyle.skip \
+       -Ddocker.tag=<TAG> \
+       -Pdocker,release              
+```
+
+When the command is finished you could find them by command `docker imaegs`.
+
+* If you want to modify DolphinScheduler source code, build and push Docker images to your registry <HUB_URL>，you can be run when finished the modification
```
Comment (kezhenxu94) [inline docs/docs/en/development/development-environment-setup.md]:
```suggestion
* If you want to modify DolphinScheduler source code, build and push Docker images to your registry <HUB_URL>，you can run when finished the modification
```

**Review by kezhenxu94 — DISMISSED** [review]
The code part looks good to me, and just review the english part of the doc, @zhongjiajie

**Review by zhongjiajie — APPROVED** [review]
LGTM , thanks

**Pull request merged**